compiler/rustc_query_impl/src/values.rs

```diff
@@ -43,3 +43,25 @@ impl<'tcx> Value<'tcx> for AdtSizedConstraint<'_> {
         }
     }
 }
+
+impl<'tcx> Value<'tcx> for ty::Binder<'_, ty::FnSig<'_>> {
+    fn from_cycle_error(tcx: QueryCtxt<'tcx>) -> Self {
+        // SAFETY: This is never called when `Self` is not `AdtSizedConstraint<'tcx>`.
+        // FIXME: Represent the above fact in the trait system somehow.
+        unsafe {
+            let err = tcx.ty_error();
+            // FIXME(compiler-errors): It would be nice if we could get the
+            // query key, so we could at least generate a fn signature that
+            // has the right arity.
+            let fn_sig = ty::Binder::dummy(tcx.mk_fn_sig(
+                [].into_iter(),
```

Comment:
Could you add a test with a function with several parameters and use it somewhere ? to verify that this line does not introduce spurious errors?